cmake/modules/libelf.cmake

```diff
@@ -5,6 +5,9 @@ option(USE_BUNDLED_LIBELF "Enable building of the bundled libelf" ${USE_BUNDLED_
 
 if(LIBELF_INCLUDE)
     # we already have LIBELF
+    # We add a custom target, in this way we can always depend on `libelf`
+    # without distinguishing between "bundled" and "not-bundled" case
+    add_custom_target(libelf)
```

Comment:
Uhm that's a good point. It shouldn't happen since in the code we cannot call more than one time `include(libelf)`, this is already true without this patch, if we call two times `include(libelf)` with `BUNDLED_LIBELF=ON` we will have an error like this

```
CMake Error at cmake/modules/libelf.cmake:10 (add_custom_target):
  add_custom_target cannot create target "libelf" because another target with the same name already exists.
```

Apart from this case, I cannot see any other valid reason to define more than one time a custom target called `libelf`, WDYT? @Molter73 @FedeDP

Reply:
> I believe cmake is complicated enough without us adding this potential pitfalls

yeah, I know, here I've tried to solve the issue we were talking about here https://github.com/falcosecurity/libs/pull/484#discussion_r925416737 :( In the `if(LIBELF_INCLUDE)` case I don't think we have other ways to create a custom target than using the `add_custom_target` explicitly :thinking:

Reply:
I actually think this might be a good opportunity to cleanup some `if(USE_BUNDLED_FOO)` around our cmake files.
@leogr wdyt?

I don't actually see the danger doing this, since we already cannot add `externalproject_add` twice.